src/main/java/cofh/thermaldynamics/util/RecipeCover.java

```diff
@@ -11,13 +11,14 @@
 import net.minecraftforge.common.ForgeHooks;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 public class RecipeCover implements IRecipe {
 
 	public static final RecipeCover instance = new RecipeCover();
 
 	@Override
-	public boolean matches(@Nonnull InventoryCrafting craft, @Nonnull World world) {
```

Comment:
Hard to say. The problem is that there's no other way to do it. It's not really a bug so much as Minecraft forcing my hand. I'm not the only mod to do something like that.